### PR TITLE
Add module source allow list

### PR DESCRIPTION
**Pull request by lonegunmanb**

### CODE REVIEW

File: container_apps/main.tf [hunk 1]
```diff
@@ -46,16 +46,24 @@ module "telemetry_proxy" {
...
+              name = "SOURCE_REGEX_${i}"
+              value = r
+            }]))
           }
         ]
       }
       ingress = {
         allow_insecure_connection = false
-        external_enabled          = true
+        external_enabled          = false
```
Comment (matt-FFFFFF) [inline container_apps/main.tf:66]: Was this intentional @lonegunmanb?

**Review by jaredfholgate — COMMENTED** [review]
It looks like the telemetry is not sent at all when the source is not matched by the regex? I'm not sure if we want that behaviour? I was thinking we just mask the source in that scenario, but still record the telemetry?

If we want to be clever we could set it to something like:

- `local file system`
- `git`
- `other`

Or just set it to something like `masked`?

I think it would still be useful to have the telemetry data for inner sourced modules, should they choose to turn it on. We just don't want to record the source uri.

@matt-FFFFFF is this your thoughts too or am I missing something?

File: container_apps/main.tf (hunk 1 above)
Comment (lonegunmanb) [reply container_apps/main.tf:66]: Yes, since now we should not enable this service without approval from AVM team, right?

**Comment by matt-FFFFFF** [comment]
> It looks like the telemetry is not sent at all when the source is not matched by the regex? I'm not sure if we want that behaviour? I was thinking we just mask the source in that scenario, but still record the telemetry?
> 
> If we want to be clever we could set it to something like:
> 
> * `local file system`
> * `git`
> * `other`
> 
> Or just set it to something like `masked`?
> 
> I think it would still be useful to have the telemetry data for inner sourced modules, should they choose to turn it on. We just don't want to record the source uri.
> 
> @matt-FFFFFF is this your thoughts too or am I missing something?

I asked the same, but if we don't have the source then what remains is a telemetry event for an unknown module. Do we want this? I think it's not that useful.

**Review by matt-FFFFFF — APPROVED** [review]

**Comment by jaredfholgate** [comment]
> > It looks like the telemetry is not sent at all when the source is not matched by the regex? I'm not sure if we want that behaviour? I was thinking we just mask the source in that scenario, but still record the telemetry?
> > If we want to be clever we could set it to something like:
> > 
> > * `local file system`
> > * `git`
> > * `other`
> > 
> > Or just set it to something like `masked`?
> > I think it would still be useful to have the telemetry data for inner sourced modules, should they choose to turn it on. We just don't want to record the source uri.
> > @matt-FFFFFF is this your thoughts too or am I missing something?
> 
> @jaredfholgate I asked the same, but if we don't have the source then what remains is a telemetry event for an unknown module. Do we want this? I think it's not that useful.

I thought we still got the module name as a separate input?

**Comment by matt-FFFFFF** [comment]
@jaredfholgate sadly not. It's sub, tenant, source, version, uuid

**Pull request merged**